tests/test_runner.py

```diff
@@ -587,3 +587,23 @@ def test_invalid_sleep():
                 with pytest.raises(type(e)):
                     runner.sleep(arg)
                 raise
+
+def test_snoop():
```

Comment:
This is pretty loose. Can you try writing a test using `check_simple`? Run some trivial code instead of this factorial example so it doesn't produce a ridiculous amount of output. If the output isn't deterministic (e.g. `0x123abc` memory addresses) try normalising it with a custom runner callback.